test/e2e/runner/rpc.go

```diff
@@ -90,22 +94,31 @@ func waitForHeight(ctx context.Context, testnet *e2e.Testnet, height int64) (*ty
 						continue
 					}
 
-					// return once all nodes have reached
-					// the target height.
-					return maxResult.Block, &maxResult.BlockID, nil
+					// All nodes are at or above the target height. Now fetch the block for that target height
+					// and return it. This will only ever be nil if the node pruned their block. Min pruning is
+					// set at least double evidence age (14) so this would mean that the node advanced more than 14
+					// blocks since we last queried it.
```

Comment:
I think we should just loop over the nodes to make it less likely we'll run into this case, instead of just using a client to the most recently queried node.

Reply:
Yeah I can make that change

Reply:
I don't see this as being a problem? we use different clients for each node and eventually get to all of the nodes.

Reply:
This loop builds a map of nodes that have all reached that target height and then checks that enough nodes have reached that height. 

Once it queries the node the last node, if this node and all other nodes had the block at the height, it gets the client of the final node it queried and tries to get a block from that node. I don't see the loop querying other nodes for the block but may be missing it? It bails out on line 93 over and over until it has seen every node.

I would maybe even restructure the loop to more explicitly separate those two actions:
One loop to check every node has a particular height
One loop to loop over all nodes to find the block.

Reply:
Yup it's sneaky but basically if the last node we picked had pruning on then it's possible that it has surpassed the specified height but no longer has the block at that height